econml/score/rscorer.py

```diff
@@ -0,0 +1,219 @@
+
+        score = 1 - loss(cate) / base_loss
+
+    This corresponds to the extra variance of the outcome explained by introducing heterogeneity
+    in the effect as captured by the cate model, as opposed to always predicting a constant effect.
+    A negative score, means that the cate model performs even worse than a constant effect model
+    and hints at overfitting during training of the cate model.
+
+    Parameters
+    ----------
+    model_y: estimator
+        The estimator for fitting the response to the features. Must implement
+        `fit` and `predict` methods.
+
+    model_t: estimator
+        The estimator for fitting the treatment to the features. Must implement
+        `fit` and `predict` methods.
+
+    discrete_treatment: bool, optional (default is ``False``)
+        Whether the treatment values should be treated as categorical, rather than continuous, quantities
+
+    categories: 'auto' or list, default 'auto'
+        The categories to use when encoding discrete treatments (or 'auto' to use the unique sorted values).
+        The first category will be treated as the control treatment.
+
+    n_splits: int, cross-validation generator or an iterable, optional (Default=2)
+        Determines the cross-validation splitting strategy.
+        Possible inputs for cv are:
+
+        - None, to use the default 3-fold cross-validation,
+        - integer, to specify the number of folds.
+        - :term:`CV splitter`
+        - An iterable yielding (train, test) splits as arrays of indices.
+
+        For integer/None inputs, if the treatment is discrete
+        :class:`~sklearn.model_selection.StratifiedKFold` is used, else,
+        :class:`~sklearn.model_selection.KFold` is used
+        (with a random shuffle in either case).
+
+        Unless an iterable is used, we call `split(concat[W, X], T)` to generate the splits. If all
+        W, X are None, then we call `split(ones((T.shape[0], 1)), T)`.
+
+    mc_iters: int, optional (default=None)
+        The number of times to rerun the first stage models to reduce the variance of the nuisances.
+
+    mc_agg: {'mean', 'median'}, optional (default='mean')
+        How to aggregate the nuisance value for each sample across the `mc_iters` monte carlo iterations of
+        cross-fitting.
+
+    random_state: int, :class:`~numpy.random.mtrand.RandomState` instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If :class:`~numpy.random.mtrand.RandomState` instance, random_state is the random number generator;
+        If None, the random number generator is the :class:`~numpy.random.mtrand.RandomState` instance used
+        by :mod:`np.random<numpy.random>`.
+    """
+
+    def __init__(self, *,
+                 model_y,
+                 model_t,
+                 discrete_treatment=False,
+                 categories='auto',
+                 n_splits=2,

```

Comment:
Why not expose `featurizer` here? I guess the init argument here is used to fit a baseline DML and get the score_ as total variance, and to compare the performance of CATE over ATE, so the more flexible to get similar nuisances with cate model the better? I know for non-DML cate model it's not possible...

Reply:
Here we are not using linear_first_stages=True. So the featurizer doesn't even enter the nusiance calculation when linear_first_sttes=False. The featurizer only affects the final model

Reply:
Okay I think I misunderstood the baseline model a bit when I left the comment. Now I think how scorer module works is by fitting a baseline model on the validation set of data, and use the MSE of training set of baseline model (which is the test set of cate model for comparison) as total variance, and use nuisances from baseline model plus the cate learnt from target model to get MSE of cate model. Do you know any reference I could take a look? especially on why we could use theta(x) learnt from model cate multiply residual t learnt from baseline model to get prediction residual y?

Reply:
I see intuitively could I think about res_y from baseline model as truth response variable on test set, and res_t from baseline as truth independent variable, and we want to learn model cate maximum r score for a given test set X.

Reply:
https://arxiv.org/pdf/1804.05146.pdf

Reply:
We are implementin the tau_risk_R in Equation (9)

Reply:
I added a ref in the docstring

Reply:
Thanks!